Tools/_sunrise/changelog/actions_changelogs_since_last_run.py

```
@@ -127,7 +383,20 @@ def get_last_changelog(changelog_file: Path | None = None) -> str:
     session.headers["X-GitHub-Api-Version"] = "2022-11-28"
 
     most_recent = get_most_recent_workflow(session, github_repository, github_run)
-    last_sha = most_recent["head_commit"]["id"]
+    if most_recent is not None:
+        title = str(most_recent.get("display_title", ""))
+        match = DISCORD_RUN_TITLE_RE.fullmatch(title)
+        if match is None:
+            raise RuntimeError("Предыдущий запуск Discord не содержит SHA опубликованного релиза")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)